rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala

```diff
@@ -4,35 +4,124 @@ import coop.rchain.rholang.syntax.rholang_mercury.Absyn.{Ground => AbsynGround,
 import org.scalatest._
 
 class BoolMatcherSpec extends FlatSpec with Matchers {
-  val matcher = BoolNormalizeMatcher
   "BoolTrue" should "Compile as GBool(true)" in {
     val btrue = new BoolTrue()
 
-    matcher.normalizeMatch(btrue) should be (GBool(true))
+    BoolNormalizeMatcher.normalizeMatch(btrue) should be (GBool(true))
   }
   "BoolFalse" should "Compile as GBool(false)" in {
     val bfalse = new BoolFalse()
 
-    matcher.normalizeMatch(bfalse) should be (GBool(false))
+    BoolNormalizeMatcher.normalizeMatch(bfalse) should be (GBool(false))
   }
 }
 
 class GroundMatcherSpec extends FlatSpec with Matchers {
-  val matcher = GroundNormalizeMatcher
   "GroundInt" should "Compile as GInt" in {
     val gi = new GroundInt(7)
 
-    matcher.normalizeMatch(gi) should be (GInt(7))
+    GroundNormalizeMatcher.normalizeMatch(gi) should be (GInt(7))
   }
   "GroundString" should "Compile as GString" in {
     val gs = new GroundString("String")
 
-    matcher.normalizeMatch(gs) should be (GString("String"))
+    GroundNormalizeMatcher.normalizeMatch(gs) should be (GString("String"))
   }
   "GroundUri" should "Compile as GUri" in {
     val gu = new GroundUri("Uri")
 
-    matcher.normalizeMatch(gu) should be (GUri("Uri"))
+    GroundNormalizeMatcher.normalizeMatch(gu) should be (GUri("Uri"))
+  }
+}
+
+class CollectMatcherSpec extends FlatSpec with Matchers {
+  val inputs = ProcVisitInputs(
+      Par(),
+      DebruijnLevelMap[VarSort]().newBindings(
+          List((Some("P"), ProcSort), (Some("x"), NameSort)))._1,
+      DebruijnLevelMap[VarSort]())
+
+  "List" should "delegate" in {
+    val listData = new ListProc()
+    listData.add(new PVar("P"))
+    listData.add(new PEval(new NameVar("x")))
+    listData.add(new PGround(new GroundInt(7)))
+    val list = new PCollect(new CollectList(listData))
+
+    val result = ProcNormalizeMatcher.normalizeMatch(list, inputs)
+    result.par should be (inputs.par.copy(
+      exprs = EList(List(
+          Par().copy(exprs = List(EVar(BoundVar(0)))),
+          Par().copy(evals = List(Eval(ChanVar(BoundVar(1))))),
+          Par().copy(exprs = List(GInt(7))))
+      ) :: inputs.par.exprs))
+    result.knownFree should be (inputs.knownFree)
+  }
+
+  "Tuple" should "delegate" in {
+    val tupleData = new ListProc()
+    tupleData.add(new PVar("Q"))
+    tupleData.add(new PEval(new NameVar("y")))
+    val tuple = new PCollect(new CollectTuple(tupleData))
+
+    val result = ProcNormalizeMatcher.normalizeMatch(tuple, inputs)
+    result.par should be (inputs.par.copy(
+      exprs = ETuple(List(
+          Par().copy(exprs = List(EVar(FreeVar(0)))),
+          Par().copy(evals = List(Eval(ChanVar(FreeVar(1))))))
+      ) :: inputs.par.exprs))
+    result.knownFree should be (inputs.knownFree.newBindings(
+        List((Some("Q"), ProcSort), (Some("y"), NameSort)))._1,
```

Comment:
The comma at the end is unnecessary. Travis complains about this in a later PR

https://travis-ci.org/rchain/rchain/jobs/343987729#L1620

Interestingly enough it works for me locally though with the comma - maybe related to how Scala implicitly tuples/untuples?